### PR TITLE
Add fail-at-first-error config setting

### PR DESCRIPTION
**Pull request by fxedel**
This PR adds the `exec-all-actions` boolean setting. When enabled, all actions are executed (even if some of them fail). If any of these actions has failed, the hook finally finishes with a non-zero exit code.

Furthermore, I adjusted the console output to better connect action logging + action errors. For CLI commands, it is now clearer which output came from which command.

### CODE REVIEW

**Comment by sebastianfeldmann** [comment]
There is too much going on in this pull request. Please split it up into multiple requests.

File: src/Runner/Hook.php [hunk 1]
```diff
@@ -114,16 +133,13 @@ public function run(): void
      */
     protected function handleAction(Config\Action $action): void
     {
-        $this->io->write(['', 'Action: <comment>' . $action->getAction() . '</comment>'], true, IO::VERBOSE);
```
Comment (sebastianfeldmann) [inline src/Runner/Hook.php:117]: Why did you remove it? Now it is duplicated in line 137 and 142.

File: src/Runner/Hook.php [hunk 2]
```diff
@@ -98,10 +99,28 @@ public function run(): void
             $this->io->write(['', '<info>No actions to execute</info>'], true, IO::VERBOSE);
             return;
         }
+
         $this->beforeHook();
+
+        $failedActions = 0;
         foreach ($actions as $action) {
-            $this->handleAction($action);
```
Comment (sebastianfeldmann) [inline src/Runner/Hook.php:103]: This nesting is a bit deep. I would propose two methods like `executeAndThrowAtFirstError` and `executeAndThrowAfterAllActions` I'm not so sure about the naming yet, but something easy to understand.

File: src/Config.php [hunk 3]
```diff
@@ -24,15 +24,16 @@
...
+    public const SETTING_BOOTSTRAP        = 'bootstrap';
+    public const SETTING_COLORS           = 'ansi-colors';
+    public const SETTING_GIT_DIR          = 'git-directory';
+    public const SETTING_INCLUDES         = 'includes';
+    public const SETTING_INCLUDES_LEVEL   = 'includes-level';
+    public const SETTING_RUN_EXEC         = 'run-exec';
+    public const SETTING_RUN_MODE         = 'run-mode';
+    public const SETTING_RUN_PATH         = 'run-path';
+    public const SETTING_VERBOSITY        = 'verbosity';
+    public const SETTING_EXEC_ALL_ACTIONS = 'exec-all-actions';
```
Comment (sebastianfeldmann) [inline src/Config.php:36]: I'm not sure about the naming.I think `FAIL_AT_FIRST_ERROR` default `true`  is more intuitive.

File: src/Config.php [hunk 4]
```diff
@@ -181,6 +182,16 @@ public function getRunPath(): string
         return (string) ($this->settings[self::SETTING_RUN_PATH] ?? '');
     }
 
+    /**
+     * Whether to always execute all actions, even if some of them have failed.
+     *
+     * @return bool
+     */
+    public function shouldExecuteAllActions(): bool
```
Comment (sebastianfeldmann) [inline src/Config.php:190]: Rename the method so it matches the constant name.

File: src/Hook/PHP/Action/Linting.php [hunk 5]
```diff
@@ -42,14 +42,20 @@ public function execute(Config $config, IO $io, Repository $repository, Config\A
     {
         $changedPHPFiles = $repository->getIndexOperator()->getStagedFilesOfType('php');
```
Comment (sebastianfeldmann) [inline src/Hook/PHP/Action/Linting.php:44]: This should not be part of this pull request

File: src/Runner/Action/Cli.php [hunk 6]
```diff
@@ -50,9 +50,19 @@ public function execute(IO $io, Repository $repository, Config\Action $action):
```
Comment (sebastianfeldmann) [inline src/Runner/Action/Cli.php:50]: This should not be part of this pull request

File: src/Runner/Action/PHP.php [hunk 7]
```diff
@@ -78,13 +78,15 @@ public function execute(Config $config, IO $io, Repository $repository, Config\A
```
Comment (sebastianfeldmann) [inline src/Runner/Action/PHP.php:78]: Should not be part of this pull request

**Review by sebastianfeldmann — CHANGES_REQUESTED** [review]

File: src/Runner/Hook.php (hunk 2 above)
Comment (sebastianfeldmann) [reply src/Runner/Hook.php:103]: I like `executeFailOnFirstError` `executeFailAfterAllAction` ;)

File: src/Runner/Hook.php (hunk 1 above)
Comment (fxedel) [reply src/Runner/Hook.php:117]: The difference is the verbosity level. I find it useful when the action name gets printed out even in normal verbosity level (so I can understand which CLI output comes from which CLI command), but skipped actions can be quiet on normal verbosity level.

Nevertheless, this more belongs to another pull request improving the console output.

**Review by sebastianfeldmann — APPROVED** [review]
Nicely done
